### DISCUSSION TITLE
❌ Advanced Display Condition by Country (Code Snippets / Sections / Popups / Widgets / Headers & Footers)

### DISCUSSION THREAD

**Discussion by RefRyan** [elementor/elementor#28196] [Feature Request]
### Prerequisites

- [ ] I have searched for similar feature requests in both open and closed discussions and cannot find an existing request.
- [ ] I have verified that the feature is still missing from the latest stable versions of Elementor or Elementor Pro.

### What problem is your feature request going to solve? Please describe..

- Being able to easily show/hide everything from code snippets, sections, popups and even headers/footers for different countries by using their IP to work out their origin country.

### Describe the solution you'd like.

A toggle to enable some advanced display conditions, that firstly asks if they want to show or hide the content because in some situations you will want to hide it for one country, in others you will only want to show it for one country.

- For Sections/Popups this would live under Advanced Rules in the Publish Settings
- For Code Snippets, this would need to be brand new UI, it would pair nicely with a change that I requested here: https://github.com/orgs/elementor/discussions/28195
- For Widgets this would live under Advanced > "Conditional Visibility" or even under the existing "Responsive" section. (Y**_ou could expand this feature to allow widgets to be shown/hidden based on browser, operating system, time/date/day (including ranges), logged in status, visit count, url paramaters, when arriving from specific URL/search engine, show x times... etc etc.. _**)

### Additional context

The detection of arriving country could be done by looking firstly for the header of CF-IPCountry which can contain their country of origin if the website is behind Cloudflare.

If the header doesn't exist then you could check their IP with one of the following services:
- KeyCDN IP Location Finder API (7.8 million requests per month; throttled to 3 requests per second)
- ip-api.com (1.9million requests per month; throttled to 45 request per minute)
- ipinfo.io (50k requests free per month, paid plans available)
- ipapi.co (30k requests free per month, paid plans available)
- ipgeolocation.io (30k requests free per month, paid plans available)
- ip2location.com (30k requests per month, paid plans available)

Alternatively the use of IP databases could be used like MaxMind GeoIP / IP2Location Lite.. etc

### Agreement

- [X] I agree that my request may be closed without action if it doesn't meet all the requirements.

**Comment by nicholaszein** [comment]
Hello there @RefRyan!

thanks for submitting your feature request! 🙏 

### 🗃️ Please understand that while we do welcome new feature requests, we can't track multiple requests per thread. 

Please follow our guidelines for submitting a new [🚀 Feature Request](https://github.com/elementor/elementor/discussions/new?category=feature-request), and submit only ☝️ **_<ins>one request per thread</ins>_** so we can track them better and allow users to vote only for the features they want to see implemented.

Kind regards
